lib/sirius_api/semester_filter.rb

```diff
@@ -0,0 +1,9 @@
+module SiriusApi
+  class SemesterFilter
+
+    def filter(dataset, params)
+      faculty_id = params[:faculty]
+      dataset.then_if(faculty_id) { |d| d.where(faculty: faculty_id) }
```

Comment:
We have `then_if` included in other module, but you should require it in the file explicitly.